rs/nns/governance/api/src/pb.rs

```diff
@@ -95,3 +113,154 @@ impl XdrConversionRate {
+                    NeuronState::NotDissolving
+                } else {
+                    NeuronState::Dissolved
+                }
+            }
+            Some(DissolveState::WhenDissolvedTimestampSeconds(
+                when_dissolved_timestamp_seconds,
+            )) => {
+                if when_dissolved_timestamp_seconds > now_seconds {
+                    NeuronState::Dissolving
+                } else {
+                    NeuronState::Dissolved
+                }
+            }
+            None => NeuronState::Dissolved,
+        }
+    }
+
+    pub fn dissolve_delay_seconds(&self, now_seconds: u64) -> u64 {
+        match self.dissolve_state {
+            Some(DissolveState::DissolveDelaySeconds(dissolve_delay_seconds)) => {
+                dissolve_delay_seconds
+            }
+            Some(DissolveState::WhenDissolvedTimestampSeconds(
+                when_dissolved_timestamp_seconds,
+            )) => when_dissolved_timestamp_seconds.saturating_sub(now_seconds),
+            None => 0,
+        }
+    }
+
+    pub fn stake_e8s(&self) -> u64 {
+        neuron_stake_e8s(
+            self.cached_neuron_stake_e8s,
+            self.neuron_fees_e8s,
+            self.staked_maturity_e8s_equivalent,
+        )
+    }
+}
+
+fn neuron_stake_e8s(
+    cached_neuron_stake_e8s: u64,
+    neuron_fees_e8s: u64,
+    staked_maturity_e8s_equivalent: Option<u64>,
+) -> u64 {
+    cached_neuron_stake_e8s
+        .saturating_sub(neuron_fees_e8s)
+        .saturating_add(staked_maturity_e8s_equivalent.unwrap_or(0))
+}
+
+impl MigrationStatus {
+    pub fn is_terminal(self) -> bool {
+        match self {
+            Self::Unspecified | Self::InProgress => false,
+            Self::Succeeded | Self::Failed => true,
+        }
+    }
+}
+
+impl CreateServiceNervousSystem {
+    pub fn sns_token_e8s(&self) -> Option<u64> {
+        self.initial_token_distribution
+            .as_ref()?
+            .swap_distribution
+            .as_ref()?
+            .total
+            .as_ref()?
+            .e8s
+    }
+
+    pub fn transaction_fee_e8s(&self) -> Option<u64> {
+        self.ledger_parameters
+            .as_ref()?
+            .transaction_fee
+            .as_ref()?
+            .e8s
+    }
+
+    pub fn neuron_minimum_stake_e8s(&self) -> Option<u64> {
+        self.governance_parameters
+            .as_ref()?
+            .neuron_minimum_stake
+            .as_ref()?
+            .e8s
+    }
+
+    /// Computes timestamps for when the SNS token swap will start, and will be
+    /// due, based on the start and end times.
+    ///
+    /// The swap will start on the first `start_time_of_day` that is more than
+    /// 24h after the swap was approved.
+    ///
+    /// The end time is calculated by adding `duration` to the computed start time.
+    ///
+    /// if start_time_of_day is None, then randomly_pick_swap_start is used to
+    /// pick a start time.
+    pub fn swap_start_and_due_timestamps(
```

Comment:
Probably missed something, but I cannot see the reason to refactor this method here.

Reply:
I don't recall, but it was in response to code not compiling.  It was used in a test somewhere that broke when I migrated the import to use ic_nns_governance_api